src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -16,40 +16,242 @@
-                    "/oauth2/fallbacks/error?error=invalid_client&error_description=Client+authentication+failed+%28e" +
-                    ".g.%2C+unknown+client%2C+no+client+authentication+included%2C+or+unsupported+authentication" +
-                    "+method%29.+The+requested+OAuth+2.0+Client+does+not+exist.";
+            throw new OAuthAuthException("invalid_client", "Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). The requested OAuth 2.0 Client does not exist.");
         } else {
             // we query hydra
+            Map<String, String> queryParamsForHydra = constructHydraRequestParamsForAuthorizationGETAPICall(paramsFromSdk);
+            Map<String, List<String>> responseHeaders = new HashMap<>();
+
+            HttpRequest.sendGETRequestWithResponseHeaders(main, "", publicOAuthProviderServiceUrl + HYDRA_AUTH_ENDPOINT, queryParamsForHydra, 10000, 10000, null, responseHeaders, false);
+
+            if(!responseHeaders.isEmpty() && responseHeaders.containsKey(LOCATION_HEADER_NAME)) {
+                String locationHeaderValue = responseHeaders.get(LOCATION_HEADER_NAME).get(0);
+                if(Utils.containsUrl(locationHeaderValue, hydraInternalAddress, true)){
+                    String error = getValueOfQueryParam(locationHeaderValue, ERROR_LITERAL);
+                    String errorDescription = getValueOfQueryParam(locationHeaderValue, ERROR_DESCRIPTION_LITERAL);
+                    throw new OAuthAuthException(error, errorDescription);
+                }
+
+                if(Utils.containsUrl(locationHeaderValue, hydraBaseUrlForConsentAndLogin, true)){
+                    redirectTo = locationHeaderValue.replace(hydraBaseUrlForConsentAndLogin, "{apiDomain}");
+                } else {
+                    redirectTo = locationHeaderValue;
+                }
+            } else {
+                throw new RuntimeException("Unexpected answer from Oauth Provider");
+            }
+            if(responseHeaders.containsKey(COOKIES_HEADER_NAME)){
+                cookies = responseHeaders.get(COOKIES_HEADER_NAME);
+            }
+        }
+
+        return new OAuthAuthResponse(redirectTo, cookies);
+    }
+
+    //This more or less acts as a pass-through for the sdks, apart from camelCase <-> snake_case key transformation and setting a few default values
+    public static JsonObject registerOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, JsonObject paramsFromSdk)
+            throws TenantOrAppNotFoundException, InvalidConfigException, IOException,
+            OAuthClientRegisterInvalidInputException,
+            NoSuchAlgorithmException, StorageQueryException {
+
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
```

Comment:
usually this will also have an API key associated with it. So that needs to go in the config as well, and used to query the admin endpoint. Please test this out in your local setup.

Reply:
Hm, I'm pretty sure that I didn't set up an API call, and yes, all my tests are using my local hydra instance. I am investigating it further and will update here.

Reply:
This: https://www.ory.sh/docs/hydra/self-hosted/production states that 
```The administrative port shouldn't be exposed to public internet traffic. If you want to expose certain endpoints, such as the /clients endpoint for OpenID Connect Dynamic Client Registry, you can do so but you need to properly secure these endpoints with an API Gateway or Authorization Proxy.```

And also:
``` None of the administrative endpoints have any built-in access control. You can do simple curl or Postman requests to talk to them.

The Token Introspection endpoint requires authentication. But since there is no access control, any valid authentication enables the endpoint to be used. If you need to access this endpoint in production, you should configure your API Gateway or Application Proxy to restrict which clients have access to the endpoint.```

Moreover, I couldn't find anything in my local setup nor in the configuration page that would suggest that hydra handles  authentication/authorization over the admin APIs.

If we need here an API key, I don't know anything about it. How will we host hydra? What kind of setup are we talking about here?

Reply:
It's ok. We can make it so that the admin endpoints are not exposed.